service/models/wishlist.py

```diff
@@ -51,52 +50,54 @@ def deserialize(self,data):
     return self
 
   def create(self):
+    """Create Wishlist instance in database"""
     logger.info("Creating %s ...", self.name)
     self.id = None
     db.session.add(self)
     db.session.commit()
     return self.id
 
   def update(self):
+    """Update Wishlist instance in database"""
     logger.info("Updating wishlist %s ...", self.name)
     if not self.id:
       raise DataValidationError("Update called with empty ID field")
     db.session.commit()
 
   def read(self):
+    """Read Wishlist instance from database"""
     logger.info("Wishlist: reading content from wishlist with id %s ...", self.id)
     wishlist = Wishlist.find_by_id(self.id)
     if wishlist is None:
       raise EntityNotFoundError("Cannot find wishlist with id {}".format(self.id))
 
-    product_ids = [wp.product_id for wp in WishlistProduct.find_all_by_wishlist_id(self.id)]
-    products = []
-    if len(product_ids) != 0:
-      products = Product.find_all_by_id(product_ids)
+    products = Product.find_all_by_wishlist_id(self.id)
     return WishlistVo(self,products).serialize()
 
   def add_products(self,product_ids:list):
```

Comment:
This method might need to go away entirely. Now to "add" and item to a wishlist we just create it with the correct wishlist_id, and can't alter it after the fact. Thoughts?

Reply:
I think may be we can use read() function in list_wishlist_by_userid() and list_all_wishlists(), which can simplify the code we already have. But it is not necessary to do that.
For example, we can change
```
  res = []
  for wishlist in wishlists:
    wishlist_products = WishlistProduct.find_all_by_wishlist_id(wishlist.id)
    products = []
    if len(wishlist_products) != 0:
      products = Product.find_all_by_id([wp.product_id for wp in wishlist_products])
    res.append(WishlistVo(wishlist,products))

  result_ser = [vo.serialize() for vo in res]
```
to
```
  res = []
  for wishlist in wishlists:
    res.append(wishlist.read())

  result_ser = res
```